test/core/end2end/fixtures/http_proxy_fixture.c

```diff
@@ -352,6 +354,42 @@ static void on_read_request_done(grpc_exec_ctx* exec_ctx, void* arg,
     GRPC_ERROR_UNREF(error);
     return;
   }
+  // If proxy auth is being used, check if the header is present
+  if(grpc_channel_args_find(
+      conn->proxy->channel_args,
+      GRPC_END2END_HTTP_PROXY_TEST_CONNECT_AUTH_PRESENT) != NULL) {
+    bool found = false, failed = false;
+    for(size_t i = 0; i < conn->http_request.hdr_count; i++) {
+      if(strcmp(conn->http_request.hdrs[i].key, "Proxy-Authorization") == 0) {
+        found = true;
+        // Authentication type should be Basic
+        if(strncmp(conn->http_request.hdrs[i].value, "Basic",
+                   strlen("Basic")) != 0) {
+          failed = true;
+          break;
+        }
+        // Check if encoded string is as expected
+        char *encoded_str_start =
```

Comment:
Check for errors. If `strchr` doesn't find a match, there'll be dereferencing a null pointer in `grpc_base64_decode`.